### PR TITLE
Feature/optional default policy

### LINKED ISSUE #14: Default policy configuration
Currently each service need to be configured in order to use rate limit
Since routes can be auto discovered in the gateway I think it will be a good idea to add a default policy configuration that can be overridden by each service

@marcosbarbero WDYT?

### PR DESCRIPTION
**Pull request by lchayoun**
@marcosbarbero WDYT?

>Fix #14

### CODE REVIEW

**Comment by marcosbarbero** [comment]
Sorry for this long delay.
This week I'm traveling and got not much available time to check this PR, will do asap.

**Comment by marcosbarbero** [comment]
I just took a look and your changes looks really nice. I'll try to merge and delivery it to maven central today.

File: spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/config/properties/RateLimitProperties.java [hunk 1]
```diff
@@ -16,35 +16,62 @@
...
+@ConfigurationProperties(RateLimitProperties.PREFIX)
 public class RateLimitProperties {
 
     public static final String PREFIX = "zuul.ratelimit";
 
-    private Map<String, Policy> policies = new LinkedHashMap<>();
-
+    private Policy defaultPolicy;
+    @NotNull
+    private Map<String, Policy> policies = Maps.newHashMap();
```
Comment (marcosbarbero) [inline spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/config/properties/RateLimitProperties.java]: Why not to use `Collections.emptyMap()` instead?

File: spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/config/properties/RateLimitProperties.java [hunk 2]
```diff
@@ -16,35 +16,62 @@
...
+    @Data
+    @NoArgsConstructor
+    public static class Policy {
+
+        @NotNull
+        private Long refreshInterval = TimeUnit.MINUTES.toSeconds(1L);
+        @NotNull
+        private Long limit;
+        @NotNull
+        private List<Type> type = Lists.newArrayList();
```
Comment (marcosbarbero) [inline spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/config/properties/RateLimitProperties.java]: Why not to use `Collections.emptyList()` instead?

File: spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/config/properties/RateLimitProperties.java (hunk 1 above)
Comment (lchayoun) [reply spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/config/properties/RateLimitProperties.java]: Spring properties is inserting to the map, not creating new one

File: spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/config/properties/RateLimitProperties.java (hunk 2 above)
Comment (lchayoun) [reply spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/config/properties/RateLimitProperties.java]: Spring properties is inserting to the map, not creating new one

**Comment by lchayoun** [comment]
@marcosbarbero will add to the readme today

**Pull request merged**